bin/index.js

```diff
@@ -1,71 +1,561 @@
+    });
+  });
+}
+
+async function getDeviceInfo(argv) {
+
+  let playerUser = players[argv.playerName].username;
+  let playerIP = players[argv.playerName].ipAddress;
+  let playerPW = players[argv.playerName].password;
+
+  let requestOptions = {
+    method: 'GET',
+    url: 'http://' + playerIP + '/api/v1/info',
+  };
   try {
-    let response = await requestAxios({method: `${requestMethod}`, url: `${requestRoute}`});
-    // console.log(JSON.stringify(response));
-  } catch (e) {
-    console.log(JSON.stringify(e));
+    let response = await requestFetch(requestOptions);
+    console.log(response);
+  } catch (err) {
+    console.log(err);
   }
+  
+
 }
 
-if (require.main === module) {
-	main();
-}
\ No newline at end of file
+async function screenshotFunc(argv) {
+
+  let playerUser = players[argv.playerName].username;
+  let playerIP = players[argv.playerName].ipAddress;
+  let playerPW = players[argv.playerName].password;
+
+  let requestOptions = {
+    method: 'POST',
+    url: 'http://' + playerIP + '/api/v1/snapshot',
+  };
+
+  try {
+    let response = await requestFetch(requestOptions);
+    //console.log(response);
+    console.log('Screenshot taken! Location: ' + response.data.result.filename);
+  } catch (err) {
+    console.log(err);
+  }
+
+
+}
+
+// General functions
+async function requestFetch(requestOptions) {
+  try {
+    let response = await fetch(requestOptions.url, requestOptions);
+    let resData = await response.json();
+    return resData;
+  } catch (err) {
+    console.error(err);
+    throw err;
+  }
+}
+
+async function checkDir(path) {
+  return new Promise((resolve, reject) => {
+    fs.stat(path, (err, stats) => {
+      if (err) {
+        reject(err);
+      }
+      if (stats.isFile()) {
+        //console.log('file');
+        resolve(true);
+      } else if (stats.isDirectory()) {
+        //console.log('directory');
+        resolve(false);
+      } else {
+        reject(new Error('Provided path is neither file nor directory'));
+      }
+    });
+  });
+}
+
+async function getFiles(path) {
+
+  try {
+    let filesArr = [];
+    let files = await fsp.readdir(path);
+
+    let lastChar = path[path.length - 1];
+    if (lastChar != '/') {
+      filesArr = files.map(file => `${path}/${file}`);
+    } else {
+      filesArr = files.map(file => `${path}${file}`);
+    }
+    return filesArr;
+  } catch (err) {
+    console.error(err);
+    throw err;
+  }
+
+}
+
+function createReadStreamFunc(path) {
+  console.log('createReadStreamFunc');
+  return new Promise((resolve, reject) => {
+    //let fileStream = createReadStreamPromise(path);
+    console.log('in promise');
+    let fileStream = fs.createReadStream(path);
+
+    fileStream.on('open', () => {
+      console.log('fileStream open');
+    });
+
+    fileStream.on('end', () => {
+      resolve(fileStream);
+      console.log('fileStream end');
+    });
+
+    fileStream.on('error', (err) => {
+      reject(err);
+    });
+
+  });
+} 
```

Comment:
This doesn't appear to be currently used. It may be left over or intended for future use - please advise.

Reply:
yeah I was trying something there that wasn't necessary - removing